lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart

```diff
@@ -665,15 +665,29 @@ class MailboxDashBoardController extends ReloadableController with UserSettingPo
 
   bool isSelectionEnabled() => currentSelectMode.value == SelectMode.ACTIVE;
 
-  void searchEmail({String? queryString}) {
-    log('MailboxDashBoardController::searchEmail():');
+  void searchEmail({
+    String? queryString,
+    EmailAddress? emailAddress
+  }) {
+    log('MailboxDashBoardController::searchEmail():QueryString = $queryString | EmailAddress = $emailAddress');
     clearFilterMessageOption();
     searchController.clearFilterSuggestion();
     if (queryString?.isNotEmpty == true) {
-      searchController.updateFilterEmail(textOption: Some(SearchQuery(queryString!)));
+      searchController.updateFilterEmail(
+        textOption: Some(SearchQuery(queryString!)),
+        sortOrderOption: searchController.sortOrderFiltered.value.getSortOrder()
+      );
+    } else {
+      searchController.updateFilterEmail(
+        sortOrderOption: searchController.sortOrderFiltered.value.getSortOrder()
+      );
+    }
+    if (emailAddress?.emailAddress.isNotEmpty == true) {
+      searchController.updateFilterEmail(toOption: Some({emailAddress!.emailAddress}));
+      dispatchAction(StartSearchEmailAction(filter: QuickSearchFilter.to));
+    } else {
+      dispatchAction(StartSearchEmailAction());
     }
```

Comment:
If both `queryString` and `emailAddress` are inputted, these if checks (`queryString?.isNotEmpty == true` and  `emailAddress?.emailAddress.isNotEmpty == true`) are both executed. Is this expected behavior? If not, consider using `else if` or `assert` statement.

Reply:
Good catch. Done